check-cla/action.yml

```diff
@@ -3,161 +3,186 @@ name: CLA check
-          console.log(hasLabel);
-          core.setOutput('hasLabel', hasLabel);
+          const has_label = labels.includes('${{ inputs.label }}')
+          core.setOutput('has_label', has_label);
+          core.debug(`has_label: ${has_label}`);
 
+    # commit status → pending
     - name: Set commit status with pending
-      uses: dholth/github-status-action@runs-using-node16
+      uses: conda/actions/set-commit-status@customize-cla-repo
       with:
-        authToken: ${{ inputs.token }}
+        token: ${{ inputs.token }}
         context: CLA check
         description: Checking conda CLA...
         state: pending
-        sha: ${{ steps.pr.outputs.sha || github.sha }}
 
+    # has_signed
     - name: Check if current actor has signed
       uses: actions/github-script@v6
       id: contributors
       with:
         github-token: ${{ inputs.token }}
         script: |
-          console.log(context);
-          const getContributors = async () => {
-            try {
-              const results = (
-                await github.rest.repos.getContent({
-                  owner: 'conda',
-                  repo: 'infrastructure',
-                  path: '.clabot'
-                })
-              );
-              return JSON.parse(Buffer.from(results.data.content, results.data.encoding).toString('utf-8')).contributors;
-            } catch (err) {
-              core.error(`Could not retrieve contributors, returning undefined. Reason: ${err}`)
-              return undefined;
-            }
-          }
-          const contributors = await getContributors();
-          console.log(contributors);
-          const pull_request = (context.payload.issue || context.payload.pull_request || context.payload);
-          const creator = pull_request.user.login;
-          console.log(creator);
-          const hasSigned = contributors.includes(creator);
-          console.log(hasSigned);
-          core.setOutput('contributors', contributors);
-          core.setOutput('hasSigned', hasSigned);
+          const { owner, repo } = context.issue;
+          core.debug(`owner: ${owner}`);
+          core.debug(`repo: ${repo}`);
+
+          const { content, encoding } = (await github.rest.repos.getContent({
+            owner: owner,
+            repo: repo,
+            path: '${{ inputs.cla_path }}'
+          })).data;
+          const contributors = JSON.parse(
+            Buffer.from(content, encoding).toString('utf-8')
+          ).contributors;
+          core.debug(`contributors: ${contributors}`);
+
+          const creator = (
+            context.payload.issue
+            || context.payload.pull_request
+            || context.payload
+          ).user.login;
+          core.debug(`creator: ${creator}`);
+
+          const has_signed = contributors.includes(creator);
+          core.setOutput('has_signed', has_signed);
+          core.debug(`has_signed: ${has_signed}`);
 
```

Comment:
Same logic but flattened since the error catching doesn't actually do anything special. Removes `contributors` output since we don't use it outside of this scope. Switch `hasSigned` to `has_signed` for variable naming consistency.

Reply:
See https://github.com/conda/actions/pull/91/files#r1158994208